cmd/zstream/zstream_decompress.c

```diff
@@ -71,8 +72,26 @@ zstream_do_decompress(int argc, char *argv[])
 
 	if (hcreate(argc) == 0)
 		errx(1, "hcreate");
-	for(i = 0; i < argc; i++) {
-		ENTRY e = {.key = argv[i]};
+	for (i = 1; i < argc; i++) {
+		int64_t inode, offset;
+		char *ino_str;
+		char *key;
+
+		key = malloc(KEYSIZE);
+
+		ino_str = strsep(&argv[i], ",");
+		if (argv[i] == NULL) {
+			zstream_usage();
+			exit(2);
+		}
+		errno = 0;
+		inode = strtoul(ino_str, NULL, 0);
+		offset = strtoul(argv[i], NULL, 0);
+		if (errno || inode < 0 || offset < 0)
+			errx(1, "invalid value for inode or offset");
+
+		snprintf(key, KEYSIZE, "%ld,%ld", inode, offset);
+		ENTRY e = {.key = key};
```

Comment:
`asprintf(&e.key, "%" PRIu64 ",%" PRIu64 "", inode, offset);` (or llu and u_ll_t, w/e) is both shorter and avoids the unchecked malloc.